doc/specs/stdlib_bitsets.md

```diff
@@ -0,0 +1,2002 @@
+equivalently be considered a sequence of logical values or a subset of
+the integers 0 ... `bits(bitset)-1`. The bits are indexed from 0 to
+`bits(bitset)-1`. A bitset is used when space savings are critical in applications that require a large number
+of closely related logical values.
+It may also improve performance by reducing memory traffic. To implement bitsets the module
+defines three bitset types, multiple constants, a character string
+literal that can be read to and from strings and formatted files, a
+simple character string literal that can be read to and from strings,
+assignments, procedures, methods, and operators. Note that the module
+assumes two's complement integers, but all current Fortran 95+ processors use such integers.
+
+
+## The module's constants
+
+The module defines several public constants all integers, almost all
+intended to serve as error codes in reporting problems through an
+optional `stat` argument. One constant, `bits_kind` is
+the integer kind value for indexing bits and reporting counts of
+bits. The other constants that are error codes are summarized below:
+
+|Error Code|Summary|
+|----------|-------|
+|`success`|No problems found|
+|`alloc_fault`|Failure with a memory allocation|
+|`array_size_invalid_error`|Attempt to define either negative bits or more than 64 bits in a `bitset_64`|
+|`char_string_invalid_error`|Invalid character found in a character string|
+|`char_string_too_large_error`|Character string was too large to be encoded in the bitset|
+|`char_string_too_small_error`|Character string was too small to hold the expected number of bits|
+|`index_invalid_error`|Index to a bitstring was less than zero or greater than the number of bits|
+|`integer_overflow_error`|Attempt to define an integer value bigger than `huge(0_bits_kind`)|
+|`read_failure`|Failure on a `read` statement|
+|`eof_failure`|An unexpected "End-of-File" on a `read` statement|
+|`write_failure`|Failure on a `write` statement|
+
+
+## The `stdlib_bitsets` derived types
+
+The `stdlib_bitsets` module defines three derived types,
+`bitset_type`, `bitset_64`, and `bitset_large`. `bitset_type` is an abstract
+type that serves as the ancestor of `bitset_64` and
+`bitset_large`. `bitset_type` defines one method, `bits`, all of its
+other methods are deferred to its extensions. `bitset_64` is a bitset
+that can handle up 64 bits. `bitset_large` is a bitset that can handle
+up `huge(0_bits_kind)` bits. All attributes of the bitset types are
+private. The various types each define a sequence of binary values: 0
+or 1. In some cases it is useful to associate a logical value, `test`,
+for each element of the sequence, where `test` is `.true.` if the value
+is 1 and `.false.` otherwise. The number of such values in an entity
+of that type is to be termed, `bits`. The bits are ordered in terms of
+position, that, in turn, is indexed from 0 to `bits-1`. `bitset_type` is
+not used in source code. The syntax for using the types are:
```

Comment:
```suggestion
not used in source code of `stdlib`, but is publicaly avaliable fo users to represent either `bitset_64` or `bitset_large`.
The syntax for using the types are:
```
Related to [this comment](https://github.com/fortran-lang/stdlib/pull/239/files#r504891872)

Reply:
I don't think this is correct. I have changed the text to

used only as a `class` to define entities that can be either a `bitset_64` or
a `bitset_large`. The syntax for using the types are:

Committed and pushed.